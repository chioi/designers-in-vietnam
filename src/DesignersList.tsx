import React from "react";

export interface IDesigner {
  id?: string;
  name: string;
  location: string;
  urls: {
    personal: string;
    social?: string;
  };
  picture: {
    small: string;
    large: string;
  };
  tags: string[];
}

interface IDesignersListProps {
  designers: IDesigner[];
}

const DesignersList = (props: IDesignersListProps) => {
  const designerWrappers = props.designers.map((designer: IDesigner) => (
    <li key={designer.name}>{designer.name}</li>
  ));
  return <ul>{designerWrappers}</ul>;
};

export default DesignersList;
