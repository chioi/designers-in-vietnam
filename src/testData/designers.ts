import { IDesigner } from "../Designer";

export const designers: IDesigner[] = [
  {
    bio: "Mexican Engineer",
    location: "Salamanca, Mexico",
    name: "Mario Gil",
    picture: { small: "", large: "" },
    tags: [],
    title: "UX designer",
    urls: {
      personal: "www.google.com"
    }
  },{
    bio: "I decide what the internet robots speak like",
    location: "Siem Reap, Cambodia",
    name: "Chankrisna Visna",
    picture: { small: "", large: "" },
    tags: ["Writer"],
    title: "Chatbot Script Designer",
    urls: {
      personal: "www.facebook.com"
    }
  }
];

export default {
  designers
}
