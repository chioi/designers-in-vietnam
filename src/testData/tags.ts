export const tag = {
  id: "hs3l233",
  name: "UX"
};

export const writerTag = {
  id: "wwy34S21",
  name: "Writer"
};

export default [tag, writerTag];
