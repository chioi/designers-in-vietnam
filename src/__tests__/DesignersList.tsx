import React from "react";
import { create, ReactTestRenderer } from "react-test-renderer";
import DesignersList from "../DesignersList";
import {designers} from "../testData/designers";

describe("gets created", () => {
  describe("without designers", () => {
    let component: ReactTestRenderer;

    beforeAll(() => {
      component = create(<DesignersList designers={[]} />);
    });

    it("has no list elements", () => {
      expect(component.root.findAllByType("li")).toHaveLength(0);
    });

    it("matches the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe("with designers", () => {
    let component: ReactTestRenderer;

    beforeAll(() => {
      component = create(<DesignersList designers={designers} />);
    });

    it(`has ${designers.length} list elements`, () => {
      expect(component.root.findAllByProps({ designer: designers[0] })).toHaveLength(1);
    });

    it("matches the snapshot", () => {
      expect(component).toMatchSnapshot();
    });

  });
});
