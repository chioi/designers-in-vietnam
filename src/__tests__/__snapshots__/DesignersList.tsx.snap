// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets created with designers matches the snapshot 1`] = `
<div
  className="horizontal-list padding-vertical-regular"
>
  <article
    className="designer"
  >
    <div>
      <img
        alt="Mario Gil's portrait"
        className="designer-portrait"
        src=""
      />
      <h2
        className="designer-name"
      >
        Mario Gil
      </h2>
      <h3
        className="designer-title"
      >
        UX designer
      </h3>
      <p
        className="designer-bio"
      >
        Mexican Engineer
      </p>
    </div>
    <footer
      className="designer-info"
    >
      <span>
        ðŸ›µ
      </span>
      <a
        className="designer-link margin-right-regular"
      >
        Site
      </a>
      <span>
        ðŸ›µ
      </span>
      <a
        className="designer-link"
      >
        Contact
      </a>
    </footer>
  </article>
  <article
    className="designer"
  >
    <div>
      <img
        alt="Chankrisna Visna's portrait"
        className="designer-portrait"
        src=""
      />
      <h2
        className="designer-name"
      >
        Chankrisna Visna
      </h2>
      <h3
        className="designer-title"
      >
        Chatbot Script Designer
      </h3>
      <p
        className="designer-bio"
      >
        I decide what the internet robots speak like
      </p>
    </div>
    <footer
      className="designer-info"
    >
      <span>
        ðŸ›µ
      </span>
      <a
        className="designer-link margin-right-regular"
      >
        Site
      </a>
      <span>
        ðŸ›µ
      </span>
      <a
        className="designer-link"
      >
        Contact
      </a>
    </footer>
  </article>
</div>
`;

exports[`gets created without designers matches the snapshot 1`] = `
<div
  className="horizontal-list padding-vertical-regular"
/>
`;
